src/System.IO.FileSystem/tests/Performance/Perf.FileStream.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Perf_FileStream : FileSystemTest
+    {
+        private static byte[] CreateBytesToWrite(int size)
+        {
+            PerfUtils utils = new PerfUtils();
+            return Text.Encoding.UTF8.GetBytes(utils.CreateString(size));
+        }
+
+        private string CreateFile(int size)
+        {
+            PerfUtils utils = new PerfUtils();
+            string filePath = GetTestFilePath();
+            byte[] bytes = Text.Encoding.UTF8.GetBytes(utils.CreateString(size));
+            File.WriteAllBytes(filePath, bytes);
+            return filePath;
+        }
+
+        [Benchmark]
+        [InlineData(2000000)]
+        [InlineData(100000000)]
+        public void Write(int size)
+        {
+            string filePath = GetTestFilePath();
+            byte[] bytes = CreateBytesToWrite(size);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (FileStream writer = File.Create(filePath))
+                using (iteration.StartMeasurement())
+                {
+                        writer.Write(bytes, 0, size);
+                }
+                File.Delete(filePath);
+            }
+        }
+
+        [Benchmark]
+        [InlineData(2000000)]
+        [InlineData(100000000)]
+        public void Read(int size)
+        {
+            string filePath = CreateFile(size);
+            byte[] bytes = new byte[size];
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (FileStream reader = File.OpenRead(filePath))
+                using (iteration.StartMeasurement())
+                {
+                    reader.Read(bytes, 0, size);
+                }
+            }
+            File.Delete(filePath);
+        }
+    }
+}
```

Comment:
It usually makes sense that an asynchronous MethodAsync is slower than its synchronous Method counterpart, as the former usually needs to do more work to set up the asynchrony and manage it.  The benefits of the async methods are around scalability and end-to-end performance benefits to the all-up application.

As for the structure, two things:
1) I think it'd be better to structure the async one as:
```C#
await ReadAsyncCore(innerIterations, bytes, bufferSize);
...
private static async Task ReadAsyncCore(int innerIterations, byte[] bytes, int bufferSize)
{
    for (int i = 0; i < innerIterations; i++)
        await reader.ReadAsync(bytes, 0, bufferSize).ConfiguratAwait(false);
}
```
2) We should probably ensure that the FileStream is constructed with FileOptions.None in the sync case and FileOptions.Asynchronous in the asynchronous case.  Otherwise, on Windows at least, with FileOptions.Asynchronous set, Read just ends up wrapping ReadAsync, and with FileOptions.Onone, ReadAsync ends up wrapping Read.